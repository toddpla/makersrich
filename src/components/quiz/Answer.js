import React from 'react'

export default class Answer extends React.Component {

  handleClick = () => {
    this.props.handleClick(this.props.id)
  }

  render() {
    return (
      <div onClick={this.handleClick}>
        {this.props.answer}
      </div>
    )
  }
}
