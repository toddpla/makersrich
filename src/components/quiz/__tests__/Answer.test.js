import React from 'react';
import { shallow } from 'enzyme'
import Answer from '../Answer'

let wrapper, handleClick

beforeEach(function() {
  handleClick = jest.fn()
  wrapper = shallow(
    <Answer
      id={'1'}
      answer={'answer'}
      correct={true}
    />
  )
});

test('it renders Answer', () => {
  expect(wrapper).toMatchSnapshot()
})
