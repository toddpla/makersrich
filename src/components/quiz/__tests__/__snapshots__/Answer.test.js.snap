// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Answer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Answer
    answer="answer"
    correct={true}
    id="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "answer",
      "className": "quiz-answer-box",
      "id": "answer-1",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "answer",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "answer",
        "className": "quiz-answer-box",
        "id": "answer-1",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "answer",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
